src/ring_theory/polynomial_algebra.lean

```diff
@@ -37,13 +37,8 @@ open algebra.tensor_product (alg_hom_of_linear_map_tensor_product include_left)
 noncomputable theory
 
 variables (R A : Type*)
-variables [comm_ring R]
-variables [ring A] [algebra R A]
-
--- The instance set up in `data.polynomial` allows `[comm_semiring R]` and `[semiring A]`.
--- Here we provide a specialisation, as otherwise some typeclass inference problems below
--- cause deterministic timeouts. Suggestions for better fixes welcome.
-instance algebra_of_algebra' : algebra R (polynomial A) := polynomial.algebra_of_algebra
```

Comment:
Fantastic! I'm glad to see this fixed.

Reply:
It's just a duplicate instance.

https://github.com/leanprover-community/mathlib/blob/0410946159f38b48600ed3f912eb283675ae7cbe/src/data/polynomial.lean#L581

https://github.com/leanprover-community/mathlib/blob/0410946159f38b48600ed3f912eb283675ae7cbe/src/data/polynomial.lean#L1880-L1887